src/Equinor.ProCoSys.IPO.Query/GetInvitations/GetInvitationsQueryBase.cs

```diff
@@ -0,0 +1,202 @@
+                           (filter.PunchOutDates.Contains(PunchOutDateFilterType.ThisWeek) &&
+                            invitation.StartTimeUtc >= startOfThisWeekUtc && invitation.StartTimeUtc < startOfNextWeekUtc) ||
+                           (filter.PunchOutDates.Contains(PunchOutDateFilterType.NextWeek) &&
+                            invitation.StartTimeUtc >= startOfNextWeekUtc && invitation.StartTimeUtc < startOfTwoWeeksUtc)) &&
+                      (!filter.IpoStatuses.Any() ||
+                            (filter.IpoStatuses.Contains(IpoStatus.Planned) && invitation.Status == IpoStatus.Planned) ||
+                            (filter.IpoStatuses.Contains(IpoStatus.Completed) && invitation.Status == IpoStatus.Completed) ||
+                            (filter.IpoStatuses.Contains(IpoStatus.Accepted) && invitation.Status == IpoStatus.Accepted) ||
+                            (filter.IpoStatuses.Contains(IpoStatus.Canceled) && invitation.Status == IpoStatus.Canceled)) &&
+                      (string.IsNullOrEmpty(filter.IpoIdStartsWith) ||
+                            invitation.Id.ToString().StartsWith(ipoIdStartWith)) &&
+                      (string.IsNullOrEmpty(filter.CommPkgNoStartsWith) ||
+                            invitation.CommPkgs.Any(c => c.CommPkgNo.ToUpper().StartsWith(filter.CommPkgNoStartsWith.ToUpper())) ||
+                            invitation.McPkgs.Any(mc => mc.CommPkgNo.ToUpper().StartsWith(filter.CommPkgNoStartsWith.ToUpper()))) &&
+                      (string.IsNullOrEmpty(filter.McPkgNoStartsWith) ||
+                            invitation.McPkgs.Any(mc => mc.McPkgNo.ToUpper().StartsWith(filter.McPkgNoStartsWith.ToUpper()))) &&
+                      (string.IsNullOrEmpty(filter.TitleStartsWith) ||
+                            invitation.Title.ToUpper().StartsWith(filter.TitleStartsWith.ToUpper())) &&
+                      (filter.PersonOid == null ||
+                            invitation.Participants.Any(p => p.AzureOid == filter.PersonOid)) &&
+                      (filter.FunctionalRoleCode == null ||
+                       invitation.Participants.Any(p => p.FunctionalRoleCode.ToUpper() == filter.FunctionalRoleCode.ToUpper())) &&
+                      (filter.PunchOutDateFromUtc == null ||
+                            invitation.StartTimeUtc >= filter.PunchOutDateFromUtc) &&
+                      (filter.PunchOutDateToUtc == null ||
+                            invitation.EndTimeUtc <= filter.PunchOutDateToUtc) &&
+                      (filter.LastChangedAtFromUtc == null ||
+                            (invitation.ModifiedAtUtc ?? invitation.CreatedAtUtc) >= filter.LastChangedAtFromUtc) &&
+                      (filter.LastChangedAtToUtc == null ||
+                            (invitation.ModifiedAtUtc ?? invitation.CreatedAtUtc) <= filter.LastChangedAtToUtc)
+                select new InvitationDto
+                {
+                    Id = EF.Property<int>(invitation, "Id"),
```

Comment:
why not invitation.Id here?

Reply:
At one stage of my query it actually didnt work, but now it works I see (with invitation.Id)